Lib/test/test_queue.py

```diff
@@ -317,38 +317,39 @@ def test_shutdown_all_methods_in_one_thread(self):
     def test_shutdown_immediate_all_methods_in_one_thread(self):
         return self._shutdown_all_methods_in_one_thread(True)
 
-    def _write_msg_thread(self, q, n, results, delay,
-                            i_when_exec_shutdown,
-                            event_start, event_end):
-        event_start.wait()
+    def _write_msg_thread(self, q, n, results,
+                            i_when_exec_shutdown, event_shutdown,
+                            event_start):
+        put_atleast = i_when_exec_shutdown//2
         for i in range(1, n+1):
             try:
                 q.put((i, "YDLO"))
-                results.append(True)
             except self.queue.ShutDown:
                 results.append(False)
-            # triggers shutdown of queue
+                break
+
+            # Be sure that all write_threads
+            # put few items into the queue.
+            if i == put_atleast:
+                event_start.wait()
```

Comment:
Sorry if this is obvious, but why is `i == i_when_exec_shutdown//2` equivalent to "all write_threads have put few items into the queue"?

Reply:
My comment is wrong about the word `few`, it should be `a few`. So I propose: `# all write_threads have put several items into the queue`. I'll rewrite this part of the code to make it clearer.